### PR TITLE
Adds the BumpVersion for our CICD workflow

### PR DESCRIPTION
**Pull request by TheMellyBee**
## Description
Get's the bumpversion ability we had in v0 ported over to v1. This allows our github actions bump the appropriate number while keeping everything in sync

## Changes

- Changes the makefile
- Adds a config file
- Hooks it together in poetry

## Related
#688  #689 

- [ X] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by andyndang** [comment]
Task linked: [CU-2jvp3ky whylogs v1 github automation: publish artifacts automation on release](https://app.clickup.com/t/2jvp3ky)

**Comment by andyndang** [comment]
Task linked: [CU-2jvp5v4 whylogs v1 github automation: bump version](https://app.clickup.com/t/2jvp5v4)

File: python/poetry.lock [hunk 1]
```diff
@@ -1684,6 +1692,22 @@ python-versions = ">=3.6"
 [package.dependencies]
 sphinx = ">=2.0"
 
+[[package]]
+name = "sphinx-rtd-theme"
```
Comment (jamie256) [inline python/poetry.lock]: Is this required for bump2version?

**Review by jamie256 — CHANGES_REQUESTED** [review]
python-ci check is failing, can you look into this? https://github.com/whylabs/whylogs/runs/7330598765?check_suite_focus=true#step:9:22

**Review by jamie256 — APPROVED** [review]
LGTM!

**Pull request merged**